### PR TITLE
Feat: support navigation on testrail API (when they are more than 250 results)

### PR DESCRIPTION
**Pull request by trooperjey01**
Loop on _links.next until is not defined to get all results

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -72,8 +72,24 @@ const prepareRun = async (
     const { value: testsResult } = await throwOnApiError(
       testrailAPI.getTests(existingRun.id)
     );
-    const currentCaseIds =
-      testsResult?.tests?.map((test) => test.case_id) || [];
+    let currentCaseIds = testsResult?.tests?.map((test) => test.case_id) || [];
+    let missingRes = false;
```
Comment (DamianOsipiuk) [inline src/index.ts:76]: You are repeating the same logic in 3 places, which in my opinion could be easily extracted to some helper function.
Would you be able to do that?
If you need some guidance, feel free to ask.

Comment (trooperjey01) [reply src/index.ts]: Hi Damian, thanks for your answer. I try to improve my code using helper functions. Let me know if it is not yet correct.

**Review by DamianOsipiuk — APPROVED** [review]

**Comment by thomaswong-pixieset** [comment]
@trooperjey01 @DamianOsipiuk With these changes, is `runid` now required in config? I am encountering the following error when `runid` is not specified. 

```[TestRail] Sending report to TestRail failed {
  url: 'https://abc.testrail.com/index.php?/api/v2/get_tests/0',
  status: 400,
  message: { error: 'Field :run_id is not a valid test run.' }
}
ERROR The "reportTaskDone" method of the "testrail" reporter produced an uncaught error. Error details:
undefined
```

**Comment by DamianOsipiuk** [comment]
Yeah, it's an oversight as `runId` should be passed from parent scope instead of taken from config.

**Comment by trooperjey01** [comment]
@thomaswong79 my bad, sorry !  I've created a PR to fix it.

**Comment by thomaswong-pixieset** [comment]
Thanks both!

**Pull request merged**